service/history/nDCActivityReplicator.go

```diff
@@ -98,7 +98,12 @@ func (r *nDCActivityReplicatorImpl) SyncActivity(
 		RunId:      request.RunId,
 	}
 
-	context, release, err := r.historyCache.getOrCreateWorkflowExecution(ctx, namespaceID, execution)
+	context, release, err := r.historyCache.getOrCreateWorkflowExecution(
+		ctx,
+		namespaceID,
+		execution,
+		callerTypeAPI,
```

Comment:
should this be callerTypeTask?

Reply:
this is for NDC activity replication, meaning usually this API is invoked by the passive side.
for passive side, replication API calls should be placed above back ground task processing.

Reply:
Is replicatorQueueProcessor.go also only happen on passive side?

Reply:
maybe some background: target (passive) cluster will call source cluster for replication tasks, this is done by periodically polling source cluster